smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSink.java

```diff
@@ -178,10 +180,14 @@ private synchronized void reportFailure(Throwable failure) {
                         .emitter(
                                 e -> stream.send((ProducerRecord) record, ar -> handleWriteResult(ar, message, record, e)));
 
-                if (this.retries > 0) {
+                if (this.retries == Integer.MAX_VALUE) {
+                    uni = uni.onFailure(this::isRecoverable).retry()
+                        .withBackOff(Duration.ofSeconds(1), Duration.ofSeconds(20)).expireIn(deliveryTimeoutMs);
```

Comment:
That uses the new configuration. Shouldn't we check for the Kafka delivery timeout?